app/views/handlers/block.py

```diff
@@ -132,6 +132,7 @@ def _update_section_completeness(
             section_id=location.section_id,
             list_item_id=location.list_item_id,
         )
+        self.questionnaire_store_updater.update_progress_for_dependent_sections()
```

Comment:
This shouldn't be called here, this would result in progress of dependent section being revaluated multiple times.

Can we not just call this before `.save()` in block.py?. Will that case any issues?

Reply:
Updated this now 👍

Reply:
Still called in `question.pu` which is duplicate call. Also in list collector it is called inside a for loop inside `evaluate_and_update_section_status_on_list_change`, while we are here, we should move that to just before "save" in `list_action.py`.

Reply:
This should be fixed now right?